src/main/java/os/failsafe/executor/TaskRepository.java

```diff
@@ -165,40 +196,24 @@ void unlock(Task toUnLock, LocalDateTime nextPlannedExecutionTime) {
         }
     }
 
-    List<Task> findAllNotLockedOrderedByCreatedDate(Set<String> processableTasks, LocalDateTime plannedExecutionDateLessOrEquals, LocalDateTime lockTimeLessOrEqual, int limit) {
+    List<Task> findAllNotLockedOrderedByCreatedDate(Connection connection, Set<String> processableTasks, LocalDateTime plannedExecutionDateLessOrEquals, LocalDateTime lockTimeLessOrEqual, int limit) {
```

Comment:
I would use tx as name for connection if it is only applicable/useful in transactional contexts.